kbchat/kvstore_test.go

```diff
@@ -0,0 +1,102 @@
+package kbchat
+
+import (
+	"testing"
+
+	"github.com/keybase/go-keybase-chat-bot/kbchat/types/keybase1"
+	"github.com/stretchr/testify/require"
+)
+
+func contains(a []string, x string) bool {
+	for _, n := range a {
+		if x == n {
+			return true
+		}
+	}
+	return false
+}
+
+func containsKey(a []keybase1.KVListEntryKey, x string) bool {
+	for _, n := range a {
+		if x == n.EntryKey {
+			return true
+		}
+	}
+	return false
+}
+
+// test doesn't make assumption about prior state
+func TestKVStore(t *testing.T) {
+	var err error
+
+	alice, dir := testBotSetup(t, "alice")
+	channel := getTeamChatChannel(t, "acme")
+	team := channel.Name
+	defer testBotTeardown(t, alice, dir)
+
+	namespace := "_test_namespace1"
+	key := "_test_key1"
+
+	// put with default revision
+	put, err := alice.PutEntry(team, namespace, key, "value1")
+	require.NoError(t, err)
+	require.True(t, put.Revision > 0)
+	rev := put.Revision
+
+	expectedRevision := rev + 1
+
+	// fail put (wrong revision)
+	_, err = alice.PutEntryWithRevision(team, namespace, key, "value2", expectedRevision-1)
+	require.Error(t, err)
+	require.Equal(t, RevisionErrorCode, err.(Error).Code)
+
+	// list namespaces
+	listns, err := alice.ListNamespaces(team)
+	require.NoError(t, err)
+	require.True(t, len(listns.Namespaces) > 0)
+	require.True(t, contains(listns.Namespaces, namespace))
```

Comment:
this isn't really a guarantee that nothing broke between this test run and the previous one, since the namespace could already have been there. i think there are two ways we traditionally handle something like this: (1) use a random string for the namespace (and entrykeys) so each run is totally distinct, (2) explicitly delete (and verify non-existence) of the namespace and entryKeys before starting the test. if it were my barbecue, i'd probably use a random string and add a `defer`ed call to a method in the test that deletes anything under that namespace.

Reply:
done